cjworkbench/middleware/i18n.py

```diff
@@ -10,11 +15,67 @@ def __init__(self, get_response):
     def __call__(self, request):
         # Code to be executed for each request before
         # the view (and later middleware) are called.
+        self._set_locale(request, self._decide_locale(request))
```

Comment:
It sets the current locale wherever it is needed (i.e. in the request and in django) and saves it to session if needed. We could move its contents in __call__ and nix it as a separate function, if you like.

Reply:
Oh, I see. `_set_locale()` does three things. I was confused about one of the three things.

Could you please just put the `_set_locale()` stuff right here, instead of in a separate function?

And then let's talk about the question I should have asked: why set `request.session["locale_id"]`? What problem does that solve?

Reply:
Session remembers the locale selection of non-registered users (and registered users that do not have a locale preference).

Reply:
Could you please describe a situation in which this makes a difference to a user? As in: Steps to reproduce + Expected results + Actual results if we don't set the session value?

Reply:
1. I enter at any page as a non-registered user with Accept-Language='en'
2. I choose Greek
3. I follow any link (e.g. sign up)

Expected: The new page is in Greek
Actual without session locale: The new page is in my Accept-Language locale

Reply:
But step 2 calls `POST /locale`, which sets the `request.session["locale_id"]` to Greek, doesn't it?

Reply:
Ah, sorry. I though you meant not to use session at all.

Saving locale to session in the middleware accomplishes two things:
1. Avoids parsing the Accept-Language on every request
2. Makes the `?locale=` testing switch act like a language switcher for non-registered users.
Indeed none of them is crucial

Reply:
Thanks. Let's nix it. I don't see value in either behavior.